File: Java_ChildrenCompetition/src/main/java/service/IUserService.java
package service;

import model.User;

//public interface IUserService <ID, T extends HasId<ID>>{
//    void login(String username, String password);
//    void logout();
//    T getCurrentUser();
//}

public interface IUserService {
    void login(String username, String password);
    void logout();
    User getCurrentUser();
}

File: CSharp_ChildrenCompetitionGUI/CSharp_ChildrenCompetitionGUI/CSharp_ChildrenCompetitionGUI/service/IUserService.cs
using CSharp_ChildrenCompetitionGUI.model;

namespace CSharp_ChildrenCompetitionGUI.service
{
    public interface IUserService<ID, T> where T : HasId<ID>
    {
        void login(string username, string password);
        void logout();
        T getCurrentUser();
    }
}